src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/Import.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public abstract class Import : EmbeddedObjectNode
+    {
+        public abstract EmbeddedPointerIndirectionNode<Signature> GetSignature(NodeFactory factory);
+    }
+
+    public class ModuleImport : Import
+    {
+        EmbeddedPointerIndirectionNode<Signature> _signature;
+
+        public ModuleImport()
+        {
+            _signature = new RvaEmbeddedPointerIndirectionNode<Signature>(new ReadyToRunHelperSignature(ReadyToRunHelper.READYTORUN_HELPER_Module));
+        }
+        public override bool StaticDependenciesAreComputed => true;
+
+        protected override int ClassCode => throw new NotImplementedException();
+
+        public override void EncodeData(ref ObjectDataBuilder dataBuilder, NodeFactory factory, bool relocsOnly)
+        {
+            // This needs to be an empty target pointer since it will be filled in with Module*
+            // when loaded by CoreCLR
+            dataBuilder.EmitZeroPointer();
+        }
+
+        public override EmbeddedPointerIndirectionNode<Signature> GetSignature(NodeFactory factory)
+        {
+            //
+            // Todo: Get this from factory once we decide on the API for making signatures
+            //
+            return _signature;
+        }
+
+        public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFactory factory)
+        {
+            var signature = GetSignature(factory);
+            if (signature != null)
```

Comment:
There is a case where method handles in the same compilation module are added as imports and we skip emitting signature for those.

Reply:
While I can imagine that the signature may end up being empty in some cases, I think that even in such case we require some RVA to be stored in the signature array as the CoreCLR runtime generally treats the two arrays (indirection cells / signatures) as parallel (as I believe Michal pointed out before). In light of this fact it might be easier to always treat the signature as existent (non-null) even if it ends up resolving to a zero-length embedded node.

Reply:
Ok, seems reasonable to me. I'll remove the null check.